utils/test/garden-test-utils.tsx

```diff
@@ -34,6 +34,29 @@ const customRtlRender = (ui: React.ReactElement, options?: any) =>
 const customDarkRender = (ui: React.ReactElement, options?: any) =>
   render(ui, { wrapper: DarkThemeWrapper, ...options });
 
+const MODE_TO_RENDER_FN_MAP = {
+  dark: customDarkRender,
+  light: customRender
+};
+/**
+ * A utility fn that returns the correct render function
+ * for the mode. Useful when using `it.each()`.
+ *
+ * @throws Throws an error if mode is not recognized.
+ *
+ * @param mode {string} The color mode.
+ * @returns {Function} Render function associated with the mode.
+ */
+export function getRenderFn(mode: string) {
```

Comment:
`mode` is loosely typed to avoid having to explicitly define the type for `mode` in tests.

### With strict `mode` type 🙅 

```typescript
it.each<['light' | 'dark', string]>([
      ['light', PALETTE.grey[700]],
      ['dark', PALETTE.grey[500]]
    ])('applies the default colors in "%s mode', (mode, color) => {
      const { container } = getRenderFn(mode)(<Component ... />);
      ...
})
```

### With loose `mode` type 🚀 

Less verbose and replies on type inferences. ✅ 

```typescript
it.each([
      ['light', PALETTE.grey[700]],
      ['dark', PALETTE.grey[500]]
    ])('applies the default colors in "%s mode', (mode, color) => {
      const { container } = getRenderFn(mode)(<Component ... />);
      ...
})
```